pyacq/devices/eeg_emotiv.py

```diff
@@ -0,0 +1,190 @@
+
+        self.lock = Mutex()
+        self.running = False
+
+    def run(self):
+        with self.lock:
+            self.running = True
+
+        n = 0
+        values = np.zeros(len(_channel_names), dtype=np.int64)
+        imp = np.zeros(len(_channel_names), dtype=np.float64)
+        gyro = np.zeros(2, dtype=np.int64)
+
+        while True:
+            with self.lock:
+                if not self.running:
+                    break
+
+            crypted_buffer = self.dev_handle.read(32)
+            data = self.cipher.decrypt(
+                crypted_buffer[:16]) + self.cipher.decrypt(crypted_buffer[16:])
+
+            # impedance value
+            sensor_num = data[0]
+            if sensor_num in _quality_num_to_name:
+                sensor_name = _quality_num_to_name[sensor_num]
+                if sensor_name in _channel_names:
+                    channel_index = _channel_names.index(sensor_name)
+                    imp[channel_index] = get_level(data, _quality_bits) / 540
+
+            # channel signals value
+            for c, channel_name in enumerate(_channel_names):
+                bits = _sensorBits[channel_name]
+                values[c] = get_level(data, bits)
+
+            # gyro value
+            gyro[0] = data[29] - 106  # X
+            gyro[1] = data[30] - 105  # Y
+
+            n += 1
+            self.outputs['signals'].send(n, values)
+            self.outputs['impedances'].send(n, imp)
+            self.outputs['gyro'].send(n, gyro)
+
+    def stop(self):
+        with self.lock:
+            self.running = False
+
+
+class Emotiv(Node):
+
+    """
+    Simple eeg emotiv device to access eeg, impedances and gyro data in a Node.
+
+    Reverse engineering and original crack code written by
+    Cody Brocious (http://github.com/daeken)
+    Kyle Machulis (http://github.com/qdot)
+    Many thanks for their contribution.
+
+    Emotiv USB emit 32-bytes reports at a rate of 128Hz, encrypted via AES
+    see https://github.com/qdot/emokit/blob/master/doc/emotiv_protocol.asciidoc
+    for more details
+
+    Parameters for configure():
+    ----
+    device_info :  dict containing :
+        - Path to the usb hidraw used
+        - Serial number of USB key
+    """
+    _output_specs = {'signals': dict(streamtype='analogsignal', dtype='int64',
+                                     shape=(-1, 14), sample_rate=128., timeaxis=0),
+                     'impedances': dict(streamtype='analogsignal', dtype='float64',
+                                        shape=(-1, 14), sample_rate=128., time_axis=0),
+                     'gyro': dict(streamtype='analogsignal', dtype='int64',
+                                  shape=(-1, 2), sample_rate=128., time_axis=0)
+                     }  # TODO Why we don't keep channel names ??
+
+    def __init__(self, **kargs):
+        Node.__init__(self, **kargs)
+        assert HAVE_PYCRYPTO, "Emotiv node depends on the `pycrypto` package, but it could not be imported."
```

Comment:
Is this a device that someone might try to use on a windows machine? If so you might add:

```
assert sys.platform != 'win32', "Emotiv is not supported on Windows"
```